libass/ass_fontselect.c

```diff
@@ -185,6 +190,80 @@ static void load_fonts_from_dir(ASS_Library *library, const char *dir)
         }
     }
     closedir(d);
+#else
+    // The dir name must ending with \*, D:\temp not work, D:\temp\* works well.
+    size_t dir_len = strlen(dir);
+    char* tdir = malloc(dir_len + 3);
+    strcpy(tdir, dir);
+    strcat(tdir, "\\*");
+    WIN32_FIND_DATAW wfdata;
+    HANDLE whFind;
+    wchar_t *wdir;
+    int wlen;
+    
+    wlen = MultiByteToWideChar(CP_UTF8, 0, tdir, -1, NULL, 0);
+    if (wlen > 0) {
+        wdir = (wchar_t*) malloc(sizeof(wchar_t) * wlen);
+        if (MultiByteToWideChar(CP_UTF8, 0, tdir, -1, wdir, wlen)) {
+            whFind = FindFirstFileW(wdir, &wfdata);
+            if (whFind != INVALID_HANDLE_VALUE) {
+                do {
+                    char *filename;
+                    int fn_len;
+                    fn_len = WideCharToMultiByte(CP_UTF8, 0, wfdata.cFileName, -1, NULL, 0, NULL, FALSE);
+                    if (fn_len > 0) {
+                        filename = (char*) malloc(fn_len);
+                        if (WideCharToMultiByte(CP_UTF8, 0, wfdata.cFileName, -1, filename, fn_len, NULL, FALSE)) {
+                            if (!strcmp(filename, ".") || !strcmp(filename, "..")) {
```

Comment:
This is different from the POSIX implementation which ignores all files/dirs starting with a dot. Same further below.